snuba/datasets/configuration/entity_builder.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+import logging
+import sys
+from typing import Any, Mapping, Sequence, Type
+
+from snuba.clickhouse.translators.snuba.allowed import (
+    FunctionCallMapper,
+    SubscriptableReferenceMapper,
+)
+from snuba.clickhouse.translators.snuba.mappers import (
+    FunctionNameMapper,
+    SubscriptableMapper,
+)
+from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
+from snuba.datasets.configuration.json_schema import V1_ENTITY_SCHEMA
+from snuba.datasets.configuration.loader import load_configuration_data
+from snuba.datasets.configuration.utils import parse_columns
+from snuba.datasets.entity import Entity
+from snuba.datasets.pluggable_entity import PluggableEntity
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage, get_writable_storage
+from snuba.query.processors import QueryProcessor
+from snuba.query.validation.validators import (
+    EntityRequiredColumnValidator,
+    QueryValidator,
+)
+
+__MAPPINGS_HAVE_BEEN_INITIALIZED = False
+
+_QP_MAPPING: dict[str, Type[QueryProcessor]] = {}
+
+_FUNCTION_MAPPER_MAPPING: Mapping[str, Type[FunctionCallMapper]] = {
+    "simple_func": FunctionNameMapper
+}
+
+_SUB_MAPPER_MAPPING: Mapping[str, Type[SubscriptableReferenceMapper]] = {
+    "subscriptable": SubscriptableMapper
+}
+
+_VALIDATOR_MAPPING: Mapping[str, Type[QueryValidator]] = {
+    "entity_required": EntityRequiredColumnValidator
+}
+
+logger = logging.getLogger("snuba.entity_builder")
+
+
+def _initialize_mappings() -> None:
+    # circular imports 🤪
```

Comment:
Obviously the emoji comment shouldn't stay but I'm wondering what degree of bad we consider having to do these imports in-method. I looked at doing some rearranging but that would have really blown up the scope of this PR